lib/repo/git_hooks/client/pre-commit.d/04-file_size_too_large.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env sh
+
+CHANGED_FILES=$(git diff --name-status --no-renames --diff-filter=d HEAD) # diff does not include deleted files
+MAX_FILE_SIZE=$(git show HEAD:.max_file_size)
+
+# shellcheck disable=SC2034
+echo "$CHANGED_FILES" | while read -r _status path; do
```

Comment:
I tried this out when deleting a single file:

```console
$ git rm A1/hello.py
$ git commit -m "tmp"
fatal: Not a valid object name d4375a28ae7ddb7c18c90c3cd94389381c0b2ffc
e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
... omitted ...
f30ebc7e481b7fe310982e2e69b25d3f55a35acd
.git/hooks/pre-commit.d/04-file_size_too_large.sh: line 10: [: : integer expression expected
```

I think the problem is that the loop here is still executing even when `$CHANGED_FILES` is "empty"? (Didn't look into this in detail, just flagging)

Reply:
Update: it seems this also happens when attempting an empty commit (i.e., no files changed). I assume for the same reason, though if there is an empty commit I think we can safely skip all of the checks?